### PR TITLE
Update ci to include code coverage after #1215

### PR DESCRIPTION
**Pull request by casz**
After #1215 the code coverage was not properly calculated.
Let's have jacoco aggregate the report :)

I'd like to use codecov to report diff coverage.

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: report/pom.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <artifactId>parent</artifactId>
+    <groupId>io.jenkins.configuration-as-code</groupId>
+    <version>${revision}${changelist}</version>
+  </parent>
+
+  <artifactId>report</artifactId>
```
Comment (oleg-nenashev) [inline report/pom.xml]:
```suggestion
  <artifactId>report-aggregator</artifactId>
```

**Comment by casz** [comment]
Coverage went from 80% to 26%
https://app.codacy.com/manual/casz/configuration-as-code-plugin/dashboard

File: report/pom.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            <phase>verify</phase>
+            <goals>
+              <goal>report-aggregate</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+
```
Comment (casz) [inline report/pom.xml]:
```suggestion
```

**Comment by casz** [comment]
Dropped the aggregating report.

GitHub action does not yet have an API, allowing codecov to whitelist github actions coverage reports ie. going tokenless.

So I changed travis to bionic and build with JDK 11 for faster builds.

**Comment by casz** [comment]
Hmm test coverage is still not properly calculated.
We need the aggregation

**Comment by casz** [comment]
okay plugin-pom's enable-jacoco does not behave well with multi module

**Comment by casz** [comment]
lots of derping around :D

**Pull request merged**